src/Shared/CertificateGeneration/MacOSCertificateManager.cs

```diff
@@ -138,94 +147,41 @@ internal override void CorrectCertificateState(X509Certificate2 candidate)
-            // as a "trust rule" applied to it.
-            // To remove the certificate we first need to remove the "trust rule" and then
-            // remove the certificate from the keychain.
-            // We don't care if we fail to remove the trust rule if
-            // for some reason the certificate became untrusted.
-            // Trying to remove the certificate from the keychain will fail if the certificate is
-            // trusted.
-            try
-            {
-                RemoveCertificateTrustRule(certificate);
-            }
-            catch
-            {
-            }
-
-            RemoveCertificateFromKeyChain(MacOSSystemKeyChain, certificate);
-        }
-        else
-        {
-            Log.MacOSCertificateUntrusted(GetDescription(certificate));
-        }
-    }
-
-    private static void RemoveCertificateTrustRule(X509Certificate2 certificate)
-    {
-        Log.MacOSRemoveCertificateTrustRuleStart(GetDescription(certificate));
-        var certificatePath = Path.GetTempFileName();
+        var tmpFile = Path.GetTempFileName();
         try
         {
-            var certBytes = certificate.Export(X509ContentType.Cert);
-            File.WriteAllBytes(certificatePath, certBytes);
-            var processInfo = new ProcessStartInfo(
-                MacOSRemoveCertificateTrustCommandLine,
-                string.Format(
-                    CultureInfo.InvariantCulture,
-                    MacOSRemoveCertificateTrustCommandLineArgumentsFormat,
-                    certificatePath
-                ));
-            using var process = Process.Start(processInfo);
-            process!.WaitForExit();
-            if (process.ExitCode != 0)
+            ExportCertificate(certificate, tmpFile, includePrivateKey: false, password: null, CertificateKeyExportFormat.Pem);
+
+            using var checkTrustProcess = Process.Start(new ProcessStartInfo(
+                MacOSVerifyCertificateCommandLine,
+                string.Format(CultureInfo.InvariantCulture, MacOSVerifyCertificateCommandLineArgumentsFormat, tmpFile))
             {
-                Log.MacOSRemoveCertificateTrustRuleError(process.ExitCode);
-            }
-            Log.MacOSRemoveCertificateTrustRuleEnd();
+                RedirectStandardOutput = true,
+                // Do this to avoid showing output to the console when the cert is not trusted. It is trivial to export
+                // the cert and replicate the command to see details.
+                RedirectStandardError = true,
+            });
+            checkTrustProcess!.WaitForExit();
+            return checkTrustProcess.ExitCode == 0;
         }
         finally
         {
-            try
-            {
-                if (File.Exists(certificatePath))
-                {
-                    File.Delete(certificatePath);
-                }
-            }
-            catch
+            if (File.Exists(tmpFile))
```

Comment:
The `File.Delete()` API checks if the file exists. However, `File.Exists()` does handle `null` properly whereas `File.Delete()` doesn't. If you are sure the path will be non-null and well-formed but unsure if it exists or not, then just calling `File.Delete()` is appropriate.